### ISSUE TITLE
Misc. bug: llama-finetune crashes with 'failed to allocate buffer of size 18446744073709547520' before training starts (all model precisions)

### ISSUE THREAD

**Issue by unravel1020** [ggml-org/llama.cpp#27648]
### Name and Version

```console
$ build/bin/Release/llama-cli.exe --version
version: 0.2.0-dev (build 10603, commit c060ca974)
built with MSVC 19.44.35228.0 for x64
```

Also reproduced at commit `39c59ed00` (build b10609). The affected files (`src/llama-context.cpp`, `ggml/src/ggml-opt.cpp`, `examples/training/finetune.cpp`) are unchanged between `c060ca974` and current upstream master (`c1c766da5`), so master is affected too.

### Operating systems

- [x] Windows

### Which llama.cpp modules do you know to be affected?

- [x] libllama (core library)
- [x] Other (llama-finetune — `examples/training/finetune.cpp`, new `llama_opt`-based training path)

### Command line

```shell
cmake -S . -B build -G "Visual Studio 17 2022" -A x64 -DGGML_CUDA=OFF -DGGML_VULKAN=OFF
cmake --build build --config Release --parallel

build\bin\Release\llama-finetune.exe -m <model> -f train.txt -o out.gguf -c 512 -lr 1e-4 -epochs 1 -opt adamw -t 16
```

### Problem description & steps to reproduce

`llama-finetune` crashes during the **first training-graph allocation**, before any epoch runs, with:

```
E ggml_backend_cpu_buffer_type_alloc_buffer: failed to allocate buffer of size 18446744073709547520
E alloc_tensor_range: failed to allocate CPU buffer of size 18446744073709547520
```

`18446744073709547520 == 0xFFFFFFFFFFFFF000 == -4096` (as int64). The requested size is a wrapped-around negative value, so some tensor in the training graph gets an invalid size.

Reproduced with three models / three weight precisions (`train.txt` is a small chat-format text file):

| model | precision | source |
|---|---|---|
| `stories260K.gguf` | F32 | https://huggingface.co/ggml-org/models/resolve/main/tinyllamas/stories260K.gguf |
| `Qwen2.5-0.5B-Instruct-f16.gguf` | F16 | https://huggingface.co/Qwen/Qwen2.5-0.5B-Instruct-GGUF |
| `Qwen2.5-0.5B-Instruct-q4_k_m.gguf` | Q4_K_M | same repo |

Changing `-c`, `-b`, `-ub` does not affect the failure (always size `-4096`). `test-opt.exe` passes (`46/46 tests`, `2/2 backend*optimizer`) — the ggml opt engine itself works. The crash happens right after model warmup, at the first `ggml_backend_sched_alloc_graph()` call inside `ggml_opt_alloc()` (`ggml/src/ggml-opt.cpp`), invoked from `llama_context::opt_epoch_iter` (`src/llama-context.cpp`).

### First Bad Commit

Unknown. The `llama_opt`-based finetune is new (`finetune.cpp` was rewritten from the previous `--train-data` interface; `llama_context::opt_init` even carries a `// FIXME` marker). There is currently no CI or test coverage for `llama-finetune` (no finetune invocations in `.github/workflows/` or `tests/`).

### Relevant log output

<details>
<summary>Logs</summary>

```console
0.00.039.665 E ggml_backend_cpu_buffer_type_alloc_buffer: failed to allocate buffer of size 18446744073709547520
0.00.039.669 E alloc_tensor_range: failed to allocate CPU buffer of size 18446744073709547520
```

</details>

**Comment by unravel1020** [comment]
Fixed in #27690
